datashuttle/utils_mod/rclone_utils.py

```diff
@@ -0,0 +1,166 @@
+import glob
+import os.path
+import shutil
+import subprocess
+import warnings
+from pathlib import Path
+
+from datashuttle.utils_mod import http_utils, utils
+
+
+def call_rclone(command : str, silent : bool = False):
```

Comment:
Have you tested [this](https://github.com/rclone/rclone/blob/d2fef05fe45074c458ae11d6b3361a2bd26c9c10/librclone/python/rclone.py) rather than calling a subprocess?

Reply:
unfortunately leads to some problems with compilation across systems ([discussed here](https://forum.rclone.org/t/platform-independent-use-of-librclone-python-api/33286) - the rlcone people are very helpful!).

However, this is a API that spins a HTTP server to use. This is nice as it means the job can be communicated to from other machines. However I don't know much about HTTP servers and was worried about it creating problems (vague, unspecified problems so would probably be okay) , so have avoided for now with an eye to extend in future. However, do you think it might be worth using this from the outset?

Reply:
> unfortunately leads to some problems with compilation across systems ([discussed here](https://forum.rclone.org/t/platform-independent-use-of-librclone-python-api/33286) - the rlcone people are very helpful!).

Would using conda not solve these problems?

> However, this is a API that spins a HTTP server to use. This is nice as it means the job can be communicated to from other machines. However I don't know much about HTTP servers and was worried about it creating problems (vague, unspecified problems so would probably be okay) , so have avoided for now with an eye to extend in future. However, do you think it might be worth using this from the outset?

No idea to be honest. Happy to keep it simple for now.

Reply:
I dont think so, as rather than the rclone executable the python wrapper calls librclone.so but rclone dont supply / distribute the compiled .so so it would require us to compile for all supported systems and ship them (I'm not quite sure why it does not call the rclone executable directly)